homeassistant/components/gardena_bluetooth/sensor.py

```diff
@@ -0,0 +1,135 @@
+"""Support for switch entities."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from datetime import datetime, timedelta, timezone
+
+from gardena_bluetooth.const import Battery, DeviceConfiguration, Valve
+from gardena_bluetooth.parse import Characteristic
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import Coordinator, GardenaBluetoothEntity
+
+
+@dataclass
+class GardenaBluetoothSensorEntityDescription(SensorEntityDescription):
+    """Description of entity."""
+
+    char: Characteristic = field(default_factory=lambda: Characteristic(""))
+
+
+DESCRIPTIONS = (
+    GardenaBluetoothSensorEntityDescription(
+        key=Valve.activation_reason.uuid,
+        translation_key="activation_reason",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        char=Valve.activation_reason,
+    ),
+    GardenaBluetoothSensorEntityDescription(
+        key=Battery.battery_level.uuid,
+        translation_key="battery_level",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.BATTERY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=PERCENTAGE,
+        char=Battery.battery_level,
+    ),
+    GardenaBluetoothSensorEntityDescription(
+        key=DeviceConfiguration.unix_timestamp.uuid,
+        translation_key="unix_timestamp",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        char=DeviceConfiguration.unix_timestamp,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Gardena Bluetooth sensor based on a config entry."""
+    coordinator: Coordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[GardenaBluetoothEntity] = [
+        GardenaBluetoothSensor(coordinator, description)
+        for description in DESCRIPTIONS
+        if description.key in coordinator.characteristics
```

Comment:
Should this be description.char?

Reply:
Nope. Key is a uuid which that dict is indexed by. Could have been .char.uuid which would have been the same.

Reply:
Aah, I was missing where char was used but now I see it